upstairs/src/deferred.rs

```diff
@@ -233,15 +234,39 @@ impl DeferredRead {
             let block_size = data.len() / rs.len();
             for (i, r) in rs.iter_mut().enumerate() {
                 let v = if let Some(ctx) = &self.cfg.encryption_context {
+                    let r = match r {
+                        ReadBlockContext::Empty => None,
+                        ReadBlockContext::Unencrypted { .. } => {
+                            error!(
+                                self.log,
+                                "expected encrypted but got unencrypted \
+                                 block context"
+                            );
+                            None
```

Comment:
If we return `None` here, and the `data` is all blank, will this result in
returniung `Ok(Validation::Empty)` from `validate_encrypted_read_response()`?

Reply:
Good catch, fixed in 410a87433095